pytorch_lightning/trainer/training_loop.py

```diff
@@ -92,8 +91,8 @@ def optimizer_freq_cumsum(self):
         return self._optimizer_freq_cumsum
 
     def should_skip_training(self):
-        should_by_max_steps = self.trainer.max_steps is not None and self.trainer.global_step >= self.trainer.max_steps
-        should_by_epoch = self.trainer.max_epochs is not None and self.trainer.current_epoch >= self.trainer.max_epochs
+        should_by_max_steps = self.max_steps is not None and self.global_step >= self.max_steps
+        should_by_epoch = self.max_epochs is not None and self.current_epoch >= self.max_epochs
         return should_by_max_steps or should_by_epoch or self.trainer.num_training_batches == 0
```

Comment:
super nit: bool signature
and you can return early after each line instead of handling all the ors at the end

Reply:
does Python evaluate the other side of the and if the first one is already False? I though not?

Reply:
are you asking me to do this:

```python
if condition1
    return True # return early
if condition2
    return True # return early
...
return False
```

Reply:
in anticipation of upcoming changes to the train loop I will hold back on extensive typing for now